components/request_otr/browser/request_otr_rule.cc

```diff
@@ -0,0 +1,121 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/request_otr/browser/request_otr_rule.h"
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "base/json/json_reader.h"
+#include "base/types/expected.h"
+#include "components/prefs/pref_service.h"
+#include "extensions/common/url_pattern.h"
```

Comment:
is it necessary to use `extensions/common/url_pattern` here? We are trying to move away from that because we have to hack it into android